front/src/modules/activities/components/ActivityTargetChips.tsx

```diff
@@ -12,30 +12,39 @@ const StyledContainer = styled.div`
 `;
 
 export function ActivityTargetChips({
-  targetCompanies,
-  targetPeople,
+  targets,
 }: {
-  targetCompanies?: GetCompaniesQuery;
-  targetPeople?: GetPeopleQuery;
+  targets?: Array<
+    Pick<ActivityTarget, 'id'> & {
+      person?: Pick<Person, 'id' | 'displayName' | 'avatarUrl'> | null;
+      company?: Pick<Company, 'id' | 'domainName' | 'name'> | null;
+    }
+  > | null;
 }) {
+  if (!targets) {
+    return null;
+  }
+
   return (
     <StyledContainer>
-      {targetCompanies?.companies &&
-        targetCompanies.companies.map((company) => (
+      {targets
+        .filter(({ company }) => company !== null)
+        .map(({ company }) => (
           <CompanyChip
-            key={company.id}
-            id={company.id}
-            name={company.name}
-            pictureUrl={getLogoUrlFromDomainName(company.domainName)}
+            key={company?.id}
```

Comment:
nit: company should not be nullable after your check above, we can probably find a way to tell TS that company is not null here